src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -220,38 +231,47 @@ public TypeWithAnnotations GetInferredReturnType(ConversionsBase? conversions, N
             ArrayBuilder<(BoundExpression expr, TypeWithAnnotations resultType)> returns,
             bool isAsync,
             BoundNode node,
-            ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+            ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo,
+            out bool inferredFromFunctionType)
         {
             TypeWithAnnotations bestResultType;
             int n = returns.Count;
             switch (n)
             {
                 case 0:
+                    inferredFromFunctionType = false;
                     bestResultType = default;
                     break;
                 case 1:
                     if (conversions.IncludeNullability)
                     {
+                        inferredFromFunctionType = false;
```

Comment:
I didn't follow, why is `inferredFromFunctionType` always false here?

Reply:
It's assumed that the caller has populated the `returns` array with the actual types of the corresponding `BoundExpression` (although if the caller is `NullableWalker`, the types may differ by inferred nullability). That assumption is unchanged with this PR since callers did not expect this method to return a function type previously.